code coverage workflow WIP

RaduBerinde opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108245)
<!-- Reviewable:end -->

RaduBerinde committed:
github: code coverage workflows

This change adds two GitHub Action workflows which run on each PR. One
generates unit test code coverage data, and one publishes that data to
a GCS bucket from where Reviewable can access it.

We generate coverage data using `bazel coverage`, but we restrict it
to only test the packages that have been modified by the PR.

Two workflows are required for security (the first workflow runs
potentially malicious code from a fork); for more details, see
https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

Epic: none
Release note: None

RaduBerinde committed:
codecov test PR
